allennlp/training/exponential_moving_average.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, Iterable
+from allennlp.models.model import Model
+
+
+class ExponentialMovingAverage:
```

Comment:
are there other kinds of moving averages we might care about?

should there be a `MovingAverage` base class and then the trainer can be agnostic to what type of moving average it's using, rather than hardcoding this one?

Reply:
Tensorflow as a [`moving_averages.py`](https://github.com/tensorflow/tensorflow/blob/r1.12/tensorflow/python/training/moving_averages.py) file. But, it seems that they only implemented the exponential moving average.